iphone/Maps/UI/Downloader/MWMMapDownloaderViewController.mm

```diff
@@ -124,40 +138,40 @@ - (void)searchBarCancelButtonClicked:(UISearchBar *)searchBar
 
 - (void)searchBar:(UISearchBar *)searchBar textDidChange:(NSString *)searchText
 {
-  if (searchText.length == 0)
-  {
-    self.dataSource = self.defaultDataSource;
-    [self reloadTable];
-  }
-  else
-  {
-    NSString * primaryLanguage = self.searchBar.textInputMode.primaryLanguage;
-    if (primaryLanguage)
-      m_searchParams.m_inputLocale = primaryLanguage.UTF8String;
-
-    m_searchParams.m_query = searchText.precomposedStringWithCompatibilityMapping.UTF8String;
-    GetFramework().SearchInDownloader(m_searchParams);
-  }
+  NSString * primaryLanguage = self.searchBar.textInputMode.primaryLanguage;
+  if (primaryLanguage)
+    m_searchParams.m_inputLocale = primaryLanguage.UTF8String;
+
+  m_searchParams.m_query = searchText.precomposedStringWithCompatibilityMapping.UTF8String;
+  [self updateSearchCallback];
+  GetFramework().SearchInDownloader(m_searchParams);
 }
 
 #pragma mark - UIBarPositioningDelegate
 
 - (UIBarPosition)positionForBar:(id<UIBarPositioning>)bar { return UIBarPositionTopAttached; }
 #pragma mark - Search
 
-- (void)setupSearchParams
+- (void)updateSearchCallback
 {
   __weak auto weakSelf = self;
-  m_searchParams.m_onResults = ^(DownloaderSearchResults const & results) {
+  NSTimeInterval const timestamp = [NSDate date].timeIntervalSince1970;
```

Comment:
Because of its declaration:
@property(readonly) NSTimeInterval timeIntervalSince1970;